tests/testsuite/kubevirtresource.go

```diff
@@ -198,8 +198,10 @@ func UpdateKubeVirtConfigValue(kvConfig v1.KubeVirtConfiguration) *v1.KubeVirt {
 	patch, err := strategicpatch.CreateTwoWayMergePatch(old, newJson, kv)
 	Expect(err).ToNot(HaveOccurred())
 
-	kv, err = virtClient.KubeVirt(kv.Namespace).Patch(context.Background(), kv.GetName(), types.MergePatchType, patch, metav1.PatchOptions{})
-	Expect(err).ToNot(HaveOccurred())
+	Eventually(func() error {
```

Comment:
Why is this needed?

Reply:
`Context("Errors metrics"` is a destructive test that prevents `virt-operator` from working properly to test related alerts. Sometimes this patch call can fail while `virt-operator` is not ready yet.

Reply:
What you are saying here is that due to some test, we need to retry here for this specific thing to succeed? I find it too odd and frankly worrying.

I see two options here:
- Convert this `UpdateKubeVirtConfigValue` helper to return errors instead of asserting internally. This may be harder than it looks to handle.
- At the caller of the problematic test, wait (using eventually) for something else to stabilize before proceeding to call this helper.

Reply:
I was not able to understand why 

```
12:32:01:   [FAILED] Timed out after 10.001s.
12:32:01:   Unexpected error:
12:32:01:       <*errors.errorString | 0xc0017779a0>: 
12:32:01:       failed to call healthz endpoint: Get "https://localhost:10047//healthz": EOF, component: "virt-controller", pod: "virt-controller-785cf77dbf-87488"
12:32:01:       {
12:32:01:           s: "failed to call healthz endpoint: Get \"https://localhost:10047//healthz\": EOF, component: \"virt-controller\", pod: \"virt-controller-785cf77dbf-87488\"",
12:32:01:       }
12:32:01:   occurred
12:32:01:   In [AfterEach] at: tests/tests_suite_test.go:168 @ 05/23/24 12:32:01.467
12:32:01: 
12:32:01:   Full Stack Trace
12:32:01:     kubevirt.io/kubevirt/tests_test.resetToDefaultConfig()
12:32:01:     	tests/tests_suite_test.go:168 +0x98
12:32:01:     kubevirt.io/kubevirt/tests_test.testCleanup()
12:32:01:     	tests/tests_suite_test.go:154 +0x6a
12:32:01:     kubevirt.io/kubevirt/tests_test.init.func19()
12:32:01:     	tests/tests_suite_test.go:110 +0xf
```

is happening without this improvement. Any ideas?

@EdDev

Reply:
Ohh my, based on the trace, this occurs on teardown of the whole suite [1].
When the Kubevir CR is updated, there is a mechanism in-place [2] to check it was "seen" by all other components (I find these kind of hack bad, but this is what we have :/ ). That mechanism uses the `healthz` endpoint of each component.
What we see here is that it fails to query  the virt-controller.

But this is not what you fixed now. You retried the patching, and that is not what failed from the trace you provided now.

[1] https://github.com/kubevirt/kubevirt/blob/4c9a05e593c0eb52aa6a2c12dbbb331ad52843cc/tests/tests_suite_test.go#L168
[2] https://github.com/kubevirt/kubevirt/blob/ce48d715971bcd2808c15823ea0dc329f481bee3/tests/utils.go#L707

Reply:
But I only saw this error happening without this eventually retries I think
I can try to validate the endpoint is working in the test itself, and see what happens

Reply:
Locally it seems that increasing the waiting period in the endpoint checks is enough